bin/update-packages

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+PATH = File.expand_path('../..', __FILE__)
+PACKAGES_TXT = File.join(PATH, "packages.txt")
+
+def main
+  installed = parse_package_list(`apm list --installed --bare`)
+  packages_txt = parse_package_list(File.read(PACKAGES_TXT))
+
+  packages_to_remove(packages_txt, installed).each do |name, _|
+    system("apm uninstall #{name}")
+  end
+
+  # Install only updated or missing packages
+  packages_to_install(packages_txt, installed).each do |name, version|
+    system("apm install #{name}@#{version}")
+  end
+
+  # Write new packages.txt
+  `apm list --installed --bare > #{PACKAGES_TXT}`
+
+  puts "Done!"
+end
+
+def packages_to_remove(packages_txt, installed)
+  installed.reject { |name, _| packages_txt.key?(name) }
+end
+
+def packages_to_install(packages_txt, installed)
+  packages_txt.select { |name, version|
+    !installed.key?(name) ||
+      Gem::Version.new(version) > Gem::Version.new(installed[name])
+  }
+end
+
+def parse_package_list(list)
+  list.lines
+    .map(&:strip)
+    .reject(&:empty?)
+    .map { |line| parse_package(line.strip) }
+    .to_h
```

Comment:
nevermind, I think you're right.


On Thu, Jul 9, 2015 at 10:44 AM Aaron Jensen <notifications@github.com>
wrote:

> In bin/update-packages
> <https://github.com/substantial/atomfiles/pull/6#discussion_r34282059>:
>
> > +  installed.reject { |name, _| packages_txt.key?(name) }
> > +end
> > +
> > +def packages_to_install(packages_txt, installed)
> > +  packages_txt.select { |name, version|
> > +    !installed.key?(name) ||
> > +      Gem::Version.new(version) > Gem::Version.new(installed[name])
> > +  }
> > +end
> > +
> > +def parse_package_list(list)
> > +  list.lines
> > +    .map(&:strip)
> > +    .reject(&:empty?)
> > +    .map { |line| parse_package(line.strip) }
> > +    .to_h
>
> I believe Mavericks+ is 2.0
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/substantial/atomfiles/pull/6/files#r34282059>.
>
-- 
*SUBSTANTIAL*  |  *Zachary Smith*  | Vice President, Technology |  OFFICE
: 206‑838‑0303  |  MOBILE: 206‑391‑6290